Sources/Client/Model/Channel/ChannelResponse.swift

```diff
@@ -54,8 +54,41 @@ public struct ChannelResponse: Decodable {
         if let lastMessage = messages.last,
             let messageRead = messageReads.first(where: { $0.user.isCurrent }),
             lastMessage.updated > messageRead.lastReadDate {
-            unreadMessageRead = messageRead
+            channel.unreadMessageRead = messageRead
+        }
+    }
+    
+    func calculateChannelUnreadCount() {
+        channel.unreadCountAtomic.set(0)
+        channel.mentionedUnreadCountAtomic.set(0)
+        
+        if messages.isEmpty {
+            return
         }
+        
+        var count = 0
```

Comment:
You can make `count` and `mentionedCount` `let`s since they'll be set once in `if/else`
```
let count: Int
let mentionedCount: Int
```

Reply:
I can't make it. It will rise errors.